src/functions.rs

```diff
@@ -176,6 +176,60 @@ pub(crate) fn render_servers(app: &Rc<App>) {
                 app_clone.connections.set_current(Some(addr));
                 app_clone.message_edit.set_reveal_child(false);
                 render_channels(Some(addr), &app_clone);
+
+                app_clone.connections.execute(addr, |result| {
+                    if result.is_err() { return; }
+                    let synac = result.unwrap();
+
+                    let mut channel_list: Vec<_> = synac.state.channels.values().collect();
+                    channel_list.sort_by_key(|channel| &channel.name);
+
+                    let channel_id = channel_list[0].id;
```

Comment:
Dangerous assumtion! If a channel is empty, this fails. I'd recommend using `channel_list.first()` here, which returns an optional value.

```
if let Some(channel) = channel_list.first() {
```

and then inside that block you can use `channel.id` and `channel.name`.